packages/lesswrong/server/dialogues/cron.ts

```diff
@@ -21,3 +24,81 @@ addCronJob({
     })
   }
 });
+
+const checkCompatible = (preferences:DbDialogueMatchPreference[]) => {
+  if (preferences.length >= 2) {
+
+    // Topic
+    const sourceUserYesTopics = getUserTopics(preferences[0]);
+    const targetUserYesTopics = getUserTopics(preferences[1]);
+    const sharedTopics = sourceUserYesTopics.filter(topic => targetUserYesTopics.includes(topic));
```

Comment:
If you're not actually using the `sharedTopics` for anything, and just checking whether any shared topics exist at all, this way of writing it would be clearer about the intent (and faster, though in this case the data size is small enough that it doesn't matter): `const anySharedTopics = sourceUserYesTopics.some(topic => argetUserYesTopics.includes(topic));`

`some` will short-circuit as soon as it hits an element that causes the condition to evaluate to `true`.